rust/src/operations/transaction.rs

```diff
@@ -1,194 +1,203 @@
-//! Wrapper Execution plan to handle distributed operations
-use core::any::Any;
-use std::sync::Arc;
-
-use super::*;
-use crate::action::Action;
-use crate::schema::DeltaDataTypeVersion;
-
-use arrow::array::StringArray;
-use arrow::datatypes::{
-    DataType, Field as ArrowField, Schema as ArrowSchema, SchemaRef as ArrowSchemaRef,
-};
-use arrow::record_batch::RecordBatch;
-use datafusion::{
-    execution::context::TaskContext,
-    physical_plan::{
-        coalesce_partitions::CoalescePartitionsExec, common::compute_record_batch_statistics,
-        empty::EmptyExec, expressions::PhysicalSortExpr, stream::RecordBatchStreamAdapter,
-        Distribution, ExecutionPlan, Partitioning, SendableRecordBatchStream, Statistics,
+//! Delta transactions
+use crate::action::{Action, DeltaOperation};
+use crate::storage::DeltaObjectStore;
+use crate::{crate_version, DeltaDataTypeVersion, DeltaResult, DeltaTableError};
+
+use chrono::Utc;
+use object_store::path::Path;
+use object_store::{Error as ObjectStoreError, ObjectStore};
+use serde_json::{Map, Value};
+
+const DELTA_LOG_FOLDER: &str = "_delta_log";
+
+#[derive(thiserror::Error, Debug)]
+enum TransactionError {
+    #[error("Tried committing existing table version: {0}")]
+    VersionAlreadyExists(DeltaDataTypeVersion),
+
+    /// Error returned when reading the delta log object failed.
```

Comment:
Is this description accurate? It's duplicated below, and `reading` and serializing seem like opposites.